backend/src/domain/models/Game.js

```diff
@@ -0,0 +1,46 @@
+
+/**
+ * Main game controller for the application.
+ * Controls the timing of each round and state of the game
+ */
+export class Game {
+  /**
+     * @param {*} io, server io socket connection
+     * @param {*} session, session for game to be played
+     * @param {*} swipeDeck, list of restaurants
+     */
+  constructor(io, session, swipeDeck) {
+    this.io = io;
+    this.session = session;
+    this.swipeDeck = swipeDeck;
+    this.roundInterval = session.preferences.roundInterval;
+    this.round = 0;
+  }
+
+  /**
+   * @param {*} callback, this function is called after the round interval
+   *                      has passed
+   */
+  nextRound(callback) {
+    this.round++;
+    if (this.round == this.swipeDeck.length) {
+      endGame();
+    }
+    setTimeout(
+        callback,
+        this.roundInterval*1000,
+        Date.now() + this.roundInterval*1000,
+        this.swipeDeck[round],
```

Comment:
just wondering why we need the current date + round time, as an argument in the cb?

Reply:
Its to try make sure the frontend knows exactly when the round should end.

Reply:
but we are already, setting a timer in the backend? wont that just be redundant, since we will perform the callback as soon as the timer finishes?

Reply:
It is so the front end knows exactly when the round should end and expect a response from the server. just so you dont have the problem where someone de syncs

Reply:
okay cool, sounds good